### PR TITLE
Homework5

### PR DESCRIPTION
**Pull request by norenstein8390**

### CODE REVIEW

File: src/homework5/homework5.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    hw4_report_maker = Homework4ReportMaker(df, predictors, response)
+    hw4_report_maker.make_plots_rankings()
+
+    midterm_report_maker = MidtermReportMaker(df, predictors, response)
+    midterm_report_maker.make_correlations_bruteforce()
+
+    X = df[predictors]
+    Y = df[response]
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, Y, test_size=0.2, random_state=100
```
Comment (hsinlingchen) [inline src/homework5/homework5.py]: I'm not sure what random_state=100 does to the dataset, and I wasn't able to find the meaning of it by googling. The resource I found was to do "shuffle=False, random_state=None" to make sure that it takes the historical 80% of data rows as training data and the rest 20% as testing data. But as I said I'm not entirely sure if the random_state=100 does the same, if it does, that's great!

File: src/homework5/homework5.sql [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    team_batting_counts.Double AS doubles, -- have to use team_batting_counts because 'Double' is a key word
+    Triple AS triples,
+    Home_Run AS HR,
+    Hit_By_Pitch AS HBP,
+    Sac_Fly AS SF,
+    Walk AS BB
+FROM team_batting_counts
+WHERE homeTeam = 0;
+
+CREATE INDEX away_team_batter_stats_game_id_idx ON away_team_batter_stats (game_id);
```
Comment (hsinlingchen) [inline src/homework5/homework5.sql:21]: really good that you created the index so it reduces the running time for the rest of the script. good job!

File: src/homework5/homework5.sql [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    (hp.BB + hp.H) / NULLIF(hp.IP, 0) AS home_starter_whip,
+    hp.H / NULLIF(hp.AB, 0) AS home_starter_baa,
+    hp.K AS home_starter_k,
+    ab.H / ab.AB AS away_avg,
+    (ab.H + ab.BB + ab.HBP) / (ab.AB + ab.BB + ab.HBP + ab.SF) AS away_obp,
+    (ab.singles + 2 * ab.doubles + 3 * ab.triples + 4 * ab.HR) / ab.AB AS away_slg,
+    (ap.BB + ap.H) / NULLIF(ap.IP, 0) AS away_starter_whip,
+    ap.H / NULLIF(ap.AB, 0) AS away_starter_baa,
+    ap.K AS away_starter_k,
+    CASE
```
Comment (hsinlingchen) [inline src/homework5/homework5.sql:111]: I think what you did here is great as well, to change the categorical data to 0 and 1 for the response. Great job!

File: src/homework5/homework5.sql [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+CREATE TABLE home_starting_pitcher_stats AS
+SELECT
+    game_id,
+    team_id,
+    outsPlayed / 3 AS IP,
+    Walk AS BB,
+    Hit AS H,
+    atBat AS AB,
+    Strikeout AS K
+FROM pitcher_counts
```
Comment (hsinlingchen) [inline src/homework5/homework5.sql]: maybe you could try to use team_pitching_counts table instead of pitcher_counts, I remember the professor recommended that one more than the pitcher_counts.

File: src/homework5/homework5.sql [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+CREATE TABLE away_starting_pitcher_stats AS
+SELECT
+    game_id,
+    team_id,
+    outsPlayed / 3 AS IP,
+    Walk AS BB,
+    Hit AS H,
+    atBat AS AB,
+    Strikeout AS K
+FROM pitcher_counts
```
Comment (hsinlingchen) [inline src/homework5/homework5.sql:50]: maybe you could try to use team_pitching_counts table instead of pitcher_counts, I remember the professor recommended that one more than the pitcher_counts.

File: src/homework5/homework5.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    )
+
+    decision_tree = tree.DecisionTreeClassifier()
+    decision_tree = decision_tree.fit(X_train, y_train)
+    decision_tree_score = decision_tree.score(X_test, y_test)
+    print(f"decision tree score: {decision_tree_score}")
```
Comment (hsinlingchen) [inline src/homework5/homework5.py]: it's nice to see that you took my advice from before to switch up the format of .format and f with {}. Thank you for doing that, I know it's not neccessary.

**Comment by hsinlingchen** [comment]
Overall, great job on this assignment, the model fitting scores are much higher than my own, yours are at 70% and 77%, that's really good. SQL script ran in seconds, I think the index set up really helped with the speed, also really great on that one. Plots look great, and the reports run smoothly. 
One suggestion I have is if you would like to have the report generated in the same HTML file, you can do the commend "file = open("report.html", "a")" to add to the existing report in either midterm or homework 4 report regenerating file instead of writing to a new file. But I totally understand if you want to keep them separate in the outputs, since you have two output folders created.
